### PR TITLE
22.03 Performance Release

### PR DESCRIPTION
**Pull request by ericenns**
## Description of changes
This PR started off as just fixing a scalability issue in ProjectEventHandler and has since transformed into a performance release that enables Users to have massive projects (i.e. >=100,000 samples).

The original Scalability issue that I identified was that as the number of Samples increased in a Project, certain operations would increase in time. An example is as follows:

1. Create a fresh Project
2. Create 5000 samples in that Project consecutively
   * Note: I was using API methods in [irida-uploader](https://github.com/phac-nml/irida-uploader) to do this (specifically `send_sample`);

Here are the results of the test:

| Experiment Name | IRIDA Version | IRIDA Uploader Version | Cumulative Time |
| --- | --- | ---- | --- |
| time.exp1 |  22.01.X | 0.7.1 | 5 hours and 30 minutes |
| time.exp2 | 22.03 | 0.7.1 | 20 minutes |
| time.exp3 | 22.03 | 0.8 beta | 5 minutes |

timing

Summary of Changes:
* [All]: Fixed scalability bug in updating Project modifiedDate during handling of ProjectEvent that was increasing in operation time as number of Samples increased within a project.
* [UI]: Fixed load times on Project Samples page.
* [UI]: Fixed text overflow issue in Command-line Linker modal.
* [UI]: Fixed bug with Command-line Linker modal making multiple requests to the server to get the command string.
* [UI]: Fixed issue with CSV/XLSX export on linelist page from not completing on massive projects >100,000 samples.
* [UI]: Adding in batch loading to linelist page (provides feedback to user, important on large projects >5000 samples)
* [UI]: Added a progress indicator when loading samples into the linelist table.
* [UI]: Added a warning notification when selecting all samples on the project samples page.
* [UI]: Added in missing Illumina, PacBio, and Ion Torrent instruments for NCBI submission.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/resources/ca/corefacility/bioinformatics/irida/database/changesets/22.03/add-sampleName-index-to-sample.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd">
+	<changeSet id="add-sampleName-index-to-sample" author="eric">
+		<createIndex tableName="sample" indexName="IDX_SAMPLE_SAMPLENAME">
```
Comment (deepsidhu85) [inline src/main/resources/ca/corefacility/bioinformatics/irida/database/changesets/22.03/add-sampleName-index-to-sample.xml:8]: So it looks like this breaks some stuff. If we add this changeset and try to manually load in the required-data.sql file it crashes with a weird and totally unrelated error

`bash: required-data.sql: line 2: syntax error near unexpected token `('
bash: required-data.sql: line 2: `INSERT INTO user (`createdDate`, `modifiedDate`, `email`, `firstName`, `lastName`, `locale`, `password`, `phoneNumber`, `username`, `enabled`, `system_role`, `credentialsNonExpired`) VALUES (now(), now() , 'franklin.bristow@phac-aspc.gc.ca', 'Franklin', 'Bristow', 'en', '$2a$10$yvzFLxWA9m2wNQmHpJtWT.MRZv8qV8Mo3EMB6HTkDnUbi9aBrbWWW', '7029', 'fbristow', 1, 'ROLE_USER', 1);'`

However removing the createIndex section and replacing it with this fixes it

`<sql>
CREATE UNIQUE INDEX IDX_SAMPLE_SAMPLE_NAME ON  sample(sampleName);
</sql>`

I had come across a similar issue a while back when adding in the email pipeline result stuff and ended up fixing it in the same manner.

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx [hunk 2]
```diff
@@ -1,18 +1,52 @@
-import React, { Component } from "react";
+import React from "react";
+import { Progress, Space, Typography } from "antd";
+import { Component } from "react";
+import { connect } from "react-redux";
+import { grey5 } from "../../../../../styles/colors";
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx:6]: This has been added.

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/ProjectRepositoryCustom.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+
+/**
+ * Custom repository methods for {@link Project}s
+ */
+public interface ProjectRepositoryCustom {
+
+	/**
+	 * Update the modifiedDate in a {@link Project} to the specified value.
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/ProjectRepositoryCustom.java]: Fixed in d139e72

File: src/main/resources/ca/corefacility/bioinformatics/irida/database/changesets/22.03/add-sampleName-index-to-sample.xml (hunk 1 above)
Comment (deepsidhu85) [reply src/main/resources/ca/corefacility/bioinformatics/irida/database/changesets/22.03/add-sampleName-index-to-sample.xml:8]: Thanks for taking a look and getting that fixed up

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx [hunk 4]
```diff
@@ -1,18 +1,55 @@
...
-      <div className="ag-overlay-loading-center irida-ag-overlay">
-        <div>
-          <Spin style={{ marginRight: "1rem" }} />
-          <span className="irida-ag-overlay-loading-text">{`${i18n(
-            "linelist.agGrid.loading"
-          )}`}</span>
-        </div>
+      <div
+        style={{
+          backgroundColor: "#FFFFFF",
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx:15]: Please make styled components for this `div` and the `Space` below

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx [hunk 5]
```diff
@@ -1,18 +1,55 @@
...
+            height: 300,
+            padding: 20,
+            display: "flex",
+            justifyContent: "center",
+            alignItems: "center",
+            flexDirection: "column",
+          }}
+        >
+          <Progress
+            percent={Math.floor(
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx:37]: Move this calculation into a function

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/Table.jsx [hunk 6]
```diff
@@ -213,46 +217,60 @@ export class TableComponent extends React.Component {
...
-        .filter(c => !c.hide && c.colId !== "icons");
-
-      const data = [];
-      this.api.forEachNodeAfterFilter(node => {
-        // We only need the value of the cell.
-        const item = colOrder.map(col => node.data[col.colId] || "");
-        data.push(item);
+      notification.open({
+        key: "export-notification",
+        message: "Preparing file for download",
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/linelist/components/Table/Table.jsx:230]: Localize this message and the description

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx [hunk 7]
```diff
@@ -98,13 +94,22 @@ function Linker() {
...
           </Form.Item>
         </Form>
-        <CommandText
-          className="t-cmd-text"
-          ellipsis={{ rows: 1 }}
-          copyable={command}
-        >
-          {command}
-        </CommandText>
+        <div
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx:97]: Make a styled component for this div

File: src/main/webapp/resources/js/pages/projects/share/ShareLarge.jsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { Row, Col, Alert, Progress } from "antd";
+import { shareSamplesWithProject } from "../../../apis/projects/samples";
+
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/share/ShareLarge.jsx:4]: Add file comment

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking reaallyy good! Great work @ericenns @joshsadam. Just a few small things below

File: UPGRADING.md [hunk 9]
```diff
@@ -38,9 +41,9 @@ truncate oauth_refresh_token;
...
 * This upgrade includes changes to the sample-metadata database system.  While implementing this change we encountered issues with the metadata auditing system, so we recommend additional backup steps before performing this upgrade.  See <https://phac-nml.github.io/irida-documentation/administrator/upgrades/#sample-metadata-audit-record-updates> for more details.
-* A couple new columns have been added to the announcements table in the database. The new announcement title column cannot be empty and will be given a default value. There is a script available to populate the new title from the header of the existing announcements content, which can be found under the `src/main/resources/scripts/announcements` folder in the IRIDA repo. 
+* A couple new columns have been added to the announcements table in the database. The new announcement title column cannot be empty and will be given a default value. There is a script available to populate the new title from the header of the existing announcements content, which can be found under the `src/main/resources/scripts/announcements` folder in the IRIDA repo.
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply UPGRADING.md]: Updated in 38a4c97f87c6d8ffb4c5406f98041b6bee34e371

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx:37]: Really does not need to be

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx (hunk 4 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx]: Updated in c828433cd41c82a34d019bd2377da27bb317f436

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/linelist/components/Table/LoadingOverlay.jsx]: Updated in 792c9c565c2fd496f034e075806eb56a3d4772a1

File: src/main/webapp/resources/js/pages/projects/linelist/components/Table/Table.jsx (hunk 6 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/linelist/components/Table/Table.jsx]: Updated in 2e6cbc5d33416e415106f0e72785a23ef36db19f

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx (hunk 7 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx:97]: Updated in c828433cd41c82a34d019bd2377da27bb317f436

File: src/main/webapp/resources/js/pages/projects/share/ShareLarge.jsx (hunk 8 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/share/ShareLarge.jsx:4]: Updated in 3d33ef3a2c24c2e41f437e89959bbeae2bf81cd2

**Review by deepsidhu85 — APPROVED** [review]
This looks really good. Thanks guys! That's all from me

**Pull request merged**